tcpdump-rc.rc

```diff
@@ -0,0 +1,38 @@
+#include "config.h"
+#include <winver.h>
+
+TCPDUMP_ICON ICON "tcpdump.ico"
+
+  VS_VERSION_INFO VERSIONINFO
+    FILEVERSION    PACKAGE_VERSION_WIN
+    PRODUCTVERSION PACKAGE_VERSION_WIN
+    FILEFLAGSMASK  0x3fL
+    FILEOS         VOS__WINDOWS32
+    FILETYPE       VFT_APP
+#ifdef _DEBUG
+    FILEFLAGS 0x1L
+#else
+    FILEFLAGS 0x0L
+#endif
+  BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+      BLOCK "040904b0"
+      BEGIN
+        VALUE "Comments",         "https://github.com/the-tcpdump-group/tcpdump/"
+        VALUE "CompanyName",      "The TCPdump Group"
+        VALUE "FileDescription",  "Network Monitoring and Data Acquisition Utility"
+        VALUE "FileVersion",      "PACKAGE_VERSION_WIN"
+        VALUE "InternalName",     PACKAGE_NAME
```

Comment:
IMHO, the compiler used (`"MinGW"`, `"MSVC"`) etc.) should be part of the `InternalName` or `FileDescription`.

`c:\> filever windump.exe` (from my homebuilt version):
```
  Language        0x0409 (Engelsk (USA))
  CharSet         0x04b0 Unicode
  OleSelfRegister Disabled
  CompanyName     http://www.tcpdump.org/
  FileDescription Network Packet Analyser (MSVC)
  InternalName    windump
  OriginalFilenam windump.exe
  ProductName     windump.exe
  ProductVersion  4.10.0-PRE-GIT
  FileVersion     4.10.0-PRE-GIT release
  LegalCopyright  3-clause BSD licence:
                  http://www.opensource.org/licenses/BSD-3-Clause
  PrivateBuild    Private build by <gvanem@yahoo.no>
  SpecialBuild    With: AirPcap OpenSSL SMI library
  Comments        Build on 19-November-2017
 VS_FIXEDFILEINFO:
 Signature:      feef04bd
 Struc Ver:      00010000
 FileVer:        0004000a:00000000 (4.10:0.0)
 ProdVer:        0004000a:00000000 (4.10:0.0)
 ...
```

Reply:
> IMHO, the compiler used ("MinGW", "MSVC") etc.) should be part of the InternalName or FileDescription

What do you think would be the best way to do that? Can you share a copy of your rc file?
Would be helpful if rc's and/or windres's preprocessors would define things `_MSC_VER` and `__MINGW32__`. Maybe they do. I'll try it.

>  SpecialBuild    With: AirPcap OpenSSL SMI library

A bit unrelated:
What's is the state of AirPcap? The project seems dead. And Npcap seems to have dropped it (Wireshark didn't). I also don't have a card to test on. 
Are you using winpcap or a custom build of the Packet.dll (build with `HAVE_AIRPCAP_API`)?
I was wondering because wpcap.dll (via the solution in `\Win32\Prj`) is additionally compiled with
 `winpcap\wpcap\Win32-Extensions\Win32-Extensions.c`
Clearly, that file can't be added directly to pcap's sources anymore, but `pcap_get_airpcap_handle()` in there seems needed...

Reply:
> Would be helpful if rc's and/or windres's preprocessors would define things `_MSC_VER`
> and `__MINGW32__`.

Only MinGW's `windres` does that. With MS' `rc`, one have to use `rc -nologo -D_MSC_VER`.

> What's is the state of AirPcap? The project seems dead. 

Probably. But I got a free AirPcap approx. 10 years ago. Gianluca Varenni (of CACETech) kindly sent me an adapter to help with testing. It's pretty nice even though it doesn't cover the 5GHz bands.

Reply:
> it doesn't cover the 5GHz bands

I think you mean that this limitation is hardware based. I am no AirPcap expert but it seems that the api (latest release) seems to support 5GHz bands:

```C
/*!
  \brief frequency Band.
   802.11 adapters can support different frequency bands, the most important of which are:
2.4GHz (802.11b/g/n) and 5GHz (802.11a/n).
*/
typedef enum _AirpcapChannelBand
{
    AIRPCAP_CB_AUTO = 1,				///< Automatically pick the best frequency band
    AIRPCAP_CB_2_4_GHZ = 2,				///< 2.4 GHz frequency band
    AIRPCAP_CB_4_GHZ = 4,				///< 4 GHz frequency band
    AIRPCAP_CB_5_GHZ = 5				///< 5 GHz frequency band
}AirpcapChannelBand, *PAirpcapChannelBand;
```
I am just curious because I have my own fork of the packet driver/dll and was wondering whether I should drop Airpcap and Dag support. Not if they are worth the effort... Dag probably not. but Airpcap maybe. Can you test my fork for me? I would do it myself if their cards weren't so expensive.

Reply:
> Only MinGW's windres does that. With MS' rc, one have to use rc -nologo -D_MSC_VER.

It's fine with me. I only have to differentiate between these two. So if __MINGW32__ is defined by the resource compiler's preprocessor I know it's not MSVC. If one would need to test for other compilers as well then that may be problematic (to automate). Although most of the other windows compilers are essentially gcc in disguise anyways (clang, tinycc etc.).